sonaranalyzer-dotnet/src/SonarAnalyzer.Common/Rules/PropertiesAccessCorrectFieldBase.cs

```diff
@@ -145,6 +145,16 @@ protected IList<PropertyData> CollectPropertyData(IEnumerable<IPropertySymbol> p
             return allPropertyData;
         }
 
+        protected SyntaxNode FindInvokedMethod(Compilation compilation, INamedTypeSymbol containingType, SyntaxNode expression) =>
+            compilation.GetSemanticModel(expression.SyntaxTree) is SemanticModel semanticModel
+            && semanticModel.GetSymbolInfo(expression).Symbol is ISymbol invocationSymbol
+            && invocationSymbol.ContainingType == containingType
+            && invocationSymbol.DeclaringSyntaxReferences.Length == 1
+            && invocationSymbol.DeclaringSyntaxReferences.Single().GetSyntax() is SyntaxNode invokedMethod
```

Comment:
We can use object pattern `is { }` instead of `is SemanticModel`, `is ISymbol` and `is SyntaxNode`. Specifying the same type as the method return type is misleading.